### PR TITLE
CR-1100458 multi dev fix

### PR DESCRIPTION
**Pull request by sarab96**
XMA only fix for multiple devices.
Wait for CU in separate thread

### CODE REVIEW

**Review by stsoe — CHANGES_REQUESTED** [review]
Minor changes requested

File: src/xma/include/lib/xmaapi.h [hunk 1]
```diff
@@ -90,8 +90,10 @@ typedef struct XmaSingleton
             thread1_future.wait_for(std::chrono::milliseconds(400));
         } catch (...) {}
         try {
-          if (thread2_future.valid())
-            thread2_future.wait_for(std::chrono::milliseconds(400));
+          for (auto& thread2_f: all_thread2_futures) {
+            if (thread2_f.valid())
+              thread2_f.wait_for(std::chrono::milliseconds(400));
+          }
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/include/lib/xmaapi.h]: done

File: src/xma/src/xmaapi/xmaapi.cpp [hunk 2]
```diff
@@ -455,8 +449,10 @@ void xma_exit(void)
                         g_xma_singleton->thread1_future.wait();
                 } catch (...) {}
                 try {
-                    if (g_xma_singleton->thread2_future.valid())
-                        g_xma_singleton->thread2_future.wait();
+                    for (auto& thread2_f: g_xma_singleton->all_thread2_futures) {
+                        if (thread2_f.valid())
+                          thread2_f.wait();
+                    }
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/src/xmaapi/xmaapi.cpp]: done

File: src/xma/include/lib/xmaapi.h [hunk 3]
```diff
@@ -90,8 +90,10 @@ typedef struct XmaSingleton
             thread1_future.wait_for(std::chrono::milliseconds(400));
         } catch (...) {}
         try {
-          if (thread2_future.valid())
-            thread2_future.wait_for(std::chrono::milliseconds(400));
+          for (auto& thread2_f: all_thread2_futures) {
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/include/lib/xmaapi.h]: done

**Review by stsoe — APPROVED** [review]

**Pull request merged**